lib/libc/gen/psignal.c

```diff
@@ -61,3 +63,70 @@ psiginfo(const siginfo_t *si, const char *s)
 {
 	psignal(si->si_signo, s);
 }
+
+int
+sig2str(int signum, char *str)
+{
+	int n;
+
+	if (signum <= 0 || signum > SIGRTMAX)
+		return (-1);
+
+	if (signum < sys_nsig)
+		(void) strcpy(str, sys_signame[signum]);
+	else if (signum < SIGRTMIN)
+		(void) sprintf(str, "%d", signum);
+	else {
+		if (signum <= (SIGRTMIN + SIGRTMAX) / 2) {
+			(void) strcpy(str, "RTMIN");
+			n = signum - SIGRTMIN;
+		} else {
+			(void) strcpy(str, "RTMAX");
+			n = signum - SIGRTMAX;
+		}
+		if (n != 0)
+			(void) sprintf(str + 5, "%+d", n);
+	}
+
+	return (0);
+}
+
+int
+str2sig(const char * restrict str, int * restrict pnum)
+{
+	int n, sig;
+	char *end;
+
+	if (strncmp(str, "RTMIN", 5) == 0 || strncmp(str, "RTMAX", 5) == 0) {
```

Comment:
I suggest to add
```
static const char rtmin_str[] = "RTMIN";
static const char rtmax_str[] = "RTMAX";
```
and use them and sizeof(rtmin_str) etc systematically instead of the directly placed literals and the constant '5'.

Reply:
Hopefully fixed.